app/models/manifest.rb

```diff
@@ -5,6 +5,12 @@
 class Manifest < ActiveRecord::Base
   validates :tracking_number, presence: true
 
+  before_validation :set_tracking_number, on: :create
+
+  def set_tracking_number
```

Comment:
oh yeah, in that case we'd also not have the postgres `null: false` and could add separately